src/main/java/com/conveyal/r5/streets/LinkedPointSet.java

```diff
@@ -185,10 +187,17 @@ public LinkedPointSet(LinkedPointSet sourceLinkage, WebMercatorGridPointSet subG
             for (int x = 0; x < subGrid.width; x++, pixel++) {
                 int sourceColumn = subGrid.west + x - superGrid.west;
                 int sourceRow = subGrid.north + y - superGrid.north;
-                int sourcePixel = sourceRow * superGrid.width + sourceColumn;
-                edges[pixel] = sourceLinkage.edges[sourcePixel];
-                distances0_mm[pixel] = sourceLinkage.distances0_mm[sourcePixel];
-                distances1_mm[pixel] = sourceLinkage.distances1_mm[sourcePixel];
+                if (sourceColumn < 0 || sourceColumn >= superGrid.width || sourceRow < 0 || sourceRow >= superGrid.height) { //point is outside super-grid
+                    edges[pixel] = -1; //set the edge value to -1 to indicate no linkage.
+                    distances0_mm[pixel] = MAX_OFFSTREET_WALK_METERS;
```

Comment:
It shouldn't matter what the distances are if the edges[n] == -1. The code reading it should completely skip the point and not even look at the distances.

Reply:
Also there should probably be a comment explaining that.

Reply:
I'll remove these lines

Reply:
and add a comment explaining why we don't need to set distances here.